src/sentry/api/helpers/group_index/update.py

```diff
@@ -1,7 +1,7 @@
 from __future__ import annotations
 
 from collections import defaultdict
-from typing import Any, Mapping, MutableMapping, Sequence
+from typing import Any, Dict, Mapping, MutableMapping, Sequence
```

Comment:
Same comment as https://github.com/getsentry/sentry/pull/47502#discussion_r1170264071 - supposedly `dict` is preferred.